go/libraries/doltcore/sqle/branches_table.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"io"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
+)
+
+const (
+	// BranchesTableName is the system table name
+	BranchesTableName = "dolt_branches"
+)
+
+var _ sql.Table = (*BranchesTable)(nil)
+
+// BranchesTable is a sql.Table implementation that implements a system table which shows the dolt branches
+type BranchesTable struct {
+	ddb *doltdb.DoltDB
+	rsr env.RepoStateReader
+}
+
+// NewBranchesTable creates a BranchesTable
+func NewBranchesTable(ddb *doltdb.DoltDB, rs env.RepoStateReader) *BranchesTable {
+	return &BranchesTable{ddb: ddb, rsr: rs}
+}
+
+// Name is a sql.Table interface function which returns the name of the table which is defined by the constant
+// BranchesTableName
+func (dt *BranchesTable) Name() string {
+	return BranchesTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table which is defined by the constant
+// BranchesTableName
+func (dt *BranchesTable) String() string {
+	return BranchesTableName
+}
+
+// Schema is a sql.Table interface function that gets the sql.Schema of the branches system table
+func (dt *BranchesTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "name", Type: sql.Text, Source: BranchesTableName, PrimaryKey: true},
+		{Name: "hash", Type: sql.Text, Source: BranchesTableName, PrimaryKey: true},
```

Comment:
Should this be a pkey? Aren't branch names unique?

Reply:
good catch.